pkg/reconciler/tenancy/workspacemounts/workspace_indexes.go

```diff
@@ -0,0 +1,94 @@
+/*
+Copyright 2022 The KCP Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package workspacemounts
+
+import (
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	tenancyv1alpha1 "github.com/kcp-dev/kcp/sdk/apis/tenancy/v1alpha1"
+	"github.com/kcp-dev/logicalcluster/v3"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+)
+
+const workspaceByURL = "WorkspaceByURL"
+
+// indexWorkspaceByURL is an index for workspaces by their URL.
+func indexWorkspaceByURL(obj interface{}) ([]string, error) {
+	ws, ok := obj.(*tenancyv1alpha1.Workspace)
+	if !ok {
+		return nil, fmt.Errorf("obj %T is not an Workspace", obj)
+	}
+
+	return []string{ws.Spec.URL}, nil
+}
+
+const workspaceMountsReferenceIndex = "WorkspacesByMountReference"
```

Comment:
Everything below this is copy-paste from another file to keep index-related stuff in one file.